layers/+lang/emacs-lisp/packages.el

```diff
@@ -46,6 +47,19 @@
 (defun emacs-lisp/post-init-eldoc ()
   (add-hook 'emacs-lisp-mode-hook 'eldoc-mode))
 
+(defun emacs-lisp/init-auto-compile ()
+  (use-package auto-compile
+    :defer t
+    :init
+    (progn
+      (setq auto-compile-display-buffer nil
+            auto-compile-use-mode-line t
+            auto-compile-mode-line-counter t)
```

Comment:
To be honest, it's a little awkward using use-package in spacemacs. In
someone's personal config you're probably only managing one use-package
declaration, but in spacemacs we're having to deal with people trying to
override stuff outside of the use-package declaration with the possibility
that some code is deferred.

On Thu, Oct 15, 2015 at 8:56 AM, Sebastian Wiesner <notifications@github.com
> wrote:

> In layers/+lang/emacs-lisp/packages.el
> <https://github.com/syl20bnr/spacemacs/pull/3413#discussion_r42120142>:
>
> > @@ -46,6 +47,19 @@
> >  (defun emacs-lisp/post-init-eldoc ()
> >    (add-hook 'emacs-lisp-mode-hook 'eldoc-mode))
> >
> > +(defun emacs-lisp/init-auto-compile ()
> > +  (use-package auto-compile
> > +    :defer t
> > +    :init
> > +    (progn
> > +      (setq auto-compile-display-buffer nil
> > +            auto-compile-use-mode-line t
> > +            auto-compile-mode-line-counter t)
>
> I see. I've been using :config in my old custom configuration because it
> plays better with the byte compiler checking for undefined variables, but
> you have a pretty strong point for :init. I was really just being curious [image:
> :relaxed:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/syl20bnr/spacemacs/pull/3413/files#r42120142>.
>